src/LinuxTracing/TracerThread.cpp

```diff
@@ -590,12 +617,31 @@ void TracerThread::Startup() {
 
   effective_capture_start_timestamp_ns_ = orbit_base::CaptureTimestampNs();
 
-  // Get the initial thread names and notify the listener_.
-  // All ThreadName events generate by this call will have effective_capture_start_timestamp_ns_ as
+  ModulesSnapshot modules_snapshot;
+  modules_snapshot.set_pid(target_pid_);
+  modules_snapshot.set_timestamp_ns(effective_capture_start_timestamp_ns_);
+  auto modules_or_error = orbit_elf_utils::ReadModules(target_pid_);
+  if (modules_or_error.has_value()) {
+    const std::vector<ModuleInfo>& modules = modules_or_error.value();
+    *modules_snapshot.mutable_modules() = {modules.begin(), modules.end()};
+    listener_->OnModulesSnapshot(std::move(modules_snapshot));
+  } else {
+    ERROR("Unable to load modules for %d: %s", target_pid_, modules_or_error.error().message());
+  }
+
+  // Get the initial thread names to construct CaptureStarted event and notify the listener_.
```

Comment:
"to construct CaptureStarted event" is no longer correct.

Reply:
removed.